utils/rgapi.d.ts

```diff
@@ -0,0 +1,266 @@
+ */
+export interface AppResourceResolver {
+    /**
+     * Resolves more information about an AppResource, filling in the remaining functionality of the tree item
+     * @param subContext The Azure subscription context for the AppResource
+     * @param resource The AppResource
+     */
+    resolveResource(subContext: ISubscriptionContext, resource: AppResource): vscode.ProviderResult<ResolvedAppResourceBase>;
+
+    /**
+     * Checks if this resolver is a match for a given AppResource. This should be designed to be as fast as possible.
+     * @param resource The AppResource to check if this resolver matches
+     */
+    matchesResource(resource: AppResource): boolean;
+}
+
+// Not part of public interface to start with--only Resource Groups extension will call it (for now)
+// currently implemented as AzureResourceProvider
+export interface AppResourceProvider {
+    provideResources(
+        subContext: ISubscriptionContext
+    ): vscode.ProviderResult<AppResource[]>;
+}
+
+/**
+ * A type to describe the LocalResource objects that providers should give
+ */
+export type LocalResource = AzExtTreeItem;
+
+/**
+ * A provider for supplying items for the local resources tree (e.g., storage emulator, function apps in workspace, etc.)
+ */
+export interface LocalResourceProvider {
+    /**
+     * Called to supply the tree nodes to the LocalResource tree
+     * @param parent The parent tree item (which will be the root of the local resource tree)
+     */
+    provideResources(parent: AzExtParentTreeItem): vscode.ProviderResult<LocalResource[] | undefined>;
+}
+
+/**
+ * Represents an Activity to display in the Activity Log
+ */
+export interface Activity {
+    /**
+     * An ID for the activity. Must be unique.
+     */
+    id: string;
+
+    /**
+     * If the activity supports cancellation, provide this `CancellationTokenSource`. The Activity Log will add a cancel button that will trigger this CTS.
+     */
+    cancellationTokenSource?: vscode.CancellationTokenSource;
+
+    /**
+     * Fire this event to start the activity
+     */
+    onStart: vscode.Event<OnStartActivityData>;
+
+    /**
+     * Fire this event to report progress on the activity
+     */
+    onProgress: vscode.Event<OnProgressActivityData>;
+
+    /**
+     * Fire this event when the activity succeeds
+     */
+    onSuccess: vscode.Event<OnSuccessActivityData>;
+
+    /**
+     * Fire this event when the activity fails
+     */
+    onError: vscode.Event<OnErrorActivityData>;
+}
+
+/**
+ * Options to set on the Activity log tree item
+ */
+export interface ActivityTreeItemOptions {
+    /**
+     * The label of the item
+     */
+    label: string;
+
+    /**
+     * Optionally, context values to add to the tree item. The overall context value string is a deduped, alphabetized, semicolon-separated list of individual context values.
+     */
+    contextValuesToAdd?: string[];
+
+    /**
+     * The collapsibleState of the tree item
+     * TODO: is this needed?

```

Comment:
TODO

Reply:
This is not needed anymore, there's no reason we should have to access it, and since the tree item is already created when resolving, there's no point in changing this value on resolve.